### ISSUE TITLE
Pre-training L1 Loss for Libri 960 Hours and ASR performance

### ISSUE THREAD

**Issue by Sreyan88** [s3prl/s3prl#164]
Hi,

I have been using your codebase extensively in my work. Thank You so much for this great repo!

I have been trying to use your model and pre-training from the Mockingjay paper for an ASR task.

I try pre-training on 360 hours and the L1 loss plateaus around 0.25 and the ASR performs decently. However the validation loss (libri valid clean) jumps after 0.4 when pre-training for 960 hours.

Just wanted to know if you guys have tried that earlier and the results you guys might have achieved.

Additionally if you guys have some ASR (E2E CTC Training) results for Mockingjay, would love to compare mine against yours!

Thank You

**Comment by leo19941227** [comment]
Hey!

I think @andi611 is the expert for this question :)

**Comment by andi611** [comment]
Hi,

> However the validation loss (libri valid clean) jumps after 0.4 when pre-training for 960 hours.

Since you are validating on clean speech, pre-training on 960 hours resulting in higher validation loss seems normal. The pre-training objective is reconstruction, which tends to force the model to remember everything.

In my latest work, I actually have some findings that I think you might be interested in:
We study how self-supervised models behave when pre-trained on a different amount of unlabeled data. Surprisingly, we find that methods that learn from time-only masked reconstruction methods (Mockingjay, NPC, etc.) can sometimes not benefit from more unlabeled data (960 hours). This is because of the reconstruction nature, memorizing all the details, including the unnecessary noise. 

Both NPC and Mockingjay got worse performance when we add the `train-other-500` subset for pre-train.  On the other hand, TERA too uses reconstruction loss, but can benefit from training on noisy data, thanks to the regularization provided by frequency masking.

**For more information and experimental results, please see Section V.C.2 and Fig 5. of the [TERA paper](https://arxiv.org/pdf/2007.06028.pdf).**

> Additionally if you guys have some ASR (E2E CTC Training) results for Mockingjay, would love to compare mine against yours!

We have some E2E CTC training results in the [SUPERB paper](https://arxiv.org/pdf/2105.01051.pdf), feel free to check them out!

Thanks,
Andy

**Comment by Sreyan88** [comment]
Hi @andi611 ,

Thank You for the reply, by "In my latest work" do you have this paper released?

**Comment by Sreyan88** [comment]
Thank You got it! You guys are doing great work! Mockingjay/Tera is a great replacement to contrastive learning which is dependant on large batches and exploring ways to push this is great work!

**Comment by jefflai108** [comment]
This is an interesting observation, thanks @andi611

**Comment by Sreyan88** [comment]
Hi @andi611 ,

Another question pertinent to this, have you tried pretraining Mockingjay (just Time mask kind of pre-training)for 960 hours Libri with a larger architecture?

Or in that case, have you guys tried any experiments with a larger architecture (more heads and layers) ? One thing I observe is more encoder layers and lesser attention heads degrades pre-training performance too. Any similar observation?

**Comment by andi611** [comment]
Hi @Sreyan88 

> Another question pertinent to this, have you tried pretraining Mockingjay (just Time mask kind of pre-training)for 960 hours Libri with a larger architecture?

I have tried 12-Layer Mockingjay on 360 hours, but not 960 hours.

> 
> Or in that case, have you guys tried any experiments with a larger architecture (more heads and layers) ? 

I have tried TERA (time + freq masking) with larger architectures, but not Mockingjay (time-only masking).

> One thing I observe is more encoder layers and lesser attention heads degrades pre-training performance too. Any similar observation?

Yes, I observe some similar behaviors in the TERA case. I speculate that this is because we are not providing a large enough batch size and iterations steps to suit the larger architectures.

On the other hand, with lesser attention heads the model will definitely degrade in performance too (but I understand that you need to compromise for the memory size by decreasing attention heads when you increase more layers).
